pypsa/consistency.py

```diff
@@ -314,10 +323,15 @@ def check_assets(network: Network, component: Component) -> None:
     if component.name in {"Generator", "Link"}:
         committables = network.get_committable_i(component.name)
         extendables = network.get_extendable_i(component.name)
-        intersection = committables.intersection(extendables)
+        not_mod_i = (
+            network.df(component.name)
+            .query(f"({nominal_attrs[component.name]}_mod==0)")
```

Comment:
Since this part of the code applies only to links and generators, it is safe to specify p_nom instead of using the global dictionary.
The dictionary is not needed.
Moreover, instead of using == 0, I'd recommend to compare using "p_nom>0" and adapt the code.
The reason is that >0 is fail-prone to nan values and negative values, while ==0 is not.

Reply:
I agree. Done :)